### PR TITLE
RHINENG-8201 - docker-compose based dev-setup

### PR DESCRIPTION
**Pull request by patilsuraj767**
Documentation added in [docs/dev-setup.md](https://github.com/RedHatInsights/config-manager/compare/master...patilsuraj767:config-manager:docker-compose-dev-setup?expand=1#diff-6a10aca6fb48bf9b93079bd5b76607312628c0519fca4aebfdea98fc93a99e8e)

### CODE REVIEW

File: scripts/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+      - db-cloud-connector
+  mqtt-broker:
+    image: eclipse-mosquitto
+    user: root
+    entrypoint:
+      - /bin/sh
+      - -c
+      - "/usr/sbin/mosquitto -c /mosquitto/config/mosquitto.conf"
+    volumes:
+      - "./mqtt/mosquitto:/mosquitto"
```
Comment (r14chandra) [inline scripts/docker-compose.yml:175]: Requires `:Z` else it's giving a connection refused error.

File: scripts/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    depends_on:
+      - db-cloud-connector
+      - kafka-create-topics
+      - mqtt-broker
+  kafka-consumer:
+    image: quay.io/cloudservices/cloud-connector
+    ports:
+      - "8083:8083"
+    volumes:
+      - "./client_id_to_account_id_map.json:/data/client_id_to_account_id_map.json"
```
Comment (r14chandra) [inline scripts/docker-compose.yml:209]: ditto

**Review by r14chandra — CHANGES_REQUESTED** [review]

**Review by r14chandra — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		--log-level trace \
+		--metrics-port 9007 \
+		--inventory-host http://127.0.0.1:8001 \
+		http-api
+
+start-inventory-consumer:
+	go run main.go \
+		--log-level=trace \
+		--kafka-brokers=localhost:29092 \
+		--cloud-connector-host=http://127.0.0.1:8084/api/cloud-connector/v1/ \
```
Comment (r14chandra) [inline Makefile:47]:
```suggestion
		--cloud-connector-host=http://127.0.0.1:8084/api/cloud-connector/v2/ \
```

**Review by r14chandra — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (patilsuraj767) [reply Makefile:47]: we do not need `v2` either. So removing the API version from the url.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		--cloud-connector-client-id=suraj \
+		--cloud-connector-psk=surajskey \
+		--dispatcher-psk=surajskey \
+		--playbook-host=https://random-host \
+		--metrics-port=9008 \
+		--dispatcher-host=http://127.0.0.1:8002 \
+		inventory-consumer
+
+
+configure-xjoin:
```
[the remark this answers is not in the document]
Comment (patilsuraj767) [reply Makefile:53]: Done

File: docs/dev-setup.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+Use the below make command to start inventory-consumer-service. 
+
+```bash
+make start-inventory-consumer
+```
+
+### Sending data to local config-manager. 
+```bash
+make create_host
```
Comment (r14chandra) [inline docs/dev-setup.md:57]: Should we rename it? Something like `send_mqtt_msg`?

**Review by r14chandra — CHANGES_REQUESTED** [review]
Thanks @patilsuraj767 for resolving all the comments!
I just added one suggestion.

File: docs/dev-setup.md (hunk 5 above)
Comment (patilsuraj767) [reply docs/dev-setup.md]: Done

**Review by r14chandra — APPROVED** [review]
The PR looks good. Thanks for working on it! 🚀

**Pull request merged**